src/main/java/net/minecraftforge/client/gui/GuiHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.gui;
+
+import java.util.Optional;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.container.GrindstoneContainer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.IWorldPosCallable;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.common.crafting.GrindingRecipe;
+
+/**
+ * A Class containing functions to help with gui's
+ */
+public class GuiHelper
+{
+
+    /**
+     * Adds Grinding recipes to the grinder. Takes the container, the IWorldPosCallable, the input and output inventory.
+     */
+    public static void grindingHelper(GrindstoneContainer container, IWorldPosCallable access, IInventory input, IInventory output)
+    {
+        if (input.getItem(0).isEmpty() && input.getItem(1).isEmpty())
```

Comment:
This would be an or, both need to be present for a grindstone recipe.

Reply:
But there could be recipes with only one item (like gemcutting or something)